draft-duke-quic-retry-offload.md

```diff
@@ -0,0 +1,876 @@
+client to prove ownership of its address before the server commits state.
+QUIC also has an anti-tampering mechanism to prevent the unauthorized injection
+of Retry packets into a connection. However, a server operator may want to
+offload production of Retry packets to an anti-Denial-of-Service agent or
+hardware accelerator. "Retry Offload" is a mechanism for coordination between
+a server and an external generator of Retry packets that can succeed despite
+the anti-tampering mechanism.
+
+--- middle
+
+# Introduction
+
+QUIC {{!RFC9000}} servers send Retry packets to avoid prematurely allocating
+resources when under stress, such as during a Denial of Service (DoS) attack.
+Because both Initial packets and Retry packets have weak authentication
+properties, the Retry packet contains an encrypted token that helps the client
+and server to validate, via transport parameters, that an attacker did not
+inject or modify a packet of either type for this connection attempt.
+
+However, a server under stress is less inclined to process incoming Initial
+packets and compute the Retry token in the first place. An analogous mechanism
+for TCP is syncookies {{?RFC4987}}. As TCP has weaker authentication properties
+to QUIC, syncookie generation can often be offloaded to a hardware device, or
+to a anti-Denial-of-Service provider that is topologically far from the
+protected server. As such an offload would behave exactly like an attacker,
+QUIC's authentication methods make such a capability impossible.
+
+This document seeks to enable offloading of Retry generation to QUIC via
+explicit coordination between servers and the hardware or provider offload,
+which this document refers to as a "Retry Offload." It has two different
+modes, to conform to two different use cases.
+
+The no-shared-state mode has minimal coordination and does not require key
+sharing. While operationally easier to configure and manage, it places severe
+constraints on the operational profile of the offload. In particular, the
+offload must control all ingress to the server and fail closed.
+
+The shared-state mode removes the operational constraints, but also requires
+more sophisticated key management.
+
+Both modes specify a common format for encoding information in the Retry token,
+so that the server can correctly populate the relevant transport parameter
+fields.
+
+## Terminology
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
+interpreted as described in RFC 2119 {{?RFC2119}}.
+
+In this document, these words will appear with that interpretation only when in
+ALL CAPS.  Lower case uses of these words are not to be interpreted as carrying
+significance described in RFC 2119.
+
+For brevity, "Connection ID" will often be abbreviated as "CID".
+
+A "retry offload" is a hardware or software device that is conceptually separate
```

Comment:
You use `Retry Offload` later on. We should choose a capitalization pattern and stick with it. I don't personally care what it is though.